message.go

```diff
@@ -80,6 +80,9 @@ type Message struct {
 	// A list of attachments present in the message.
 	Attachments []*MessageAttachment `json:"attachments"`
 
+	// A list of components attached to the message.
+	Components []UnmarshableMessageComponent `json:"components"`
```

Comment:
Instead of having this field be of type `UnmarshableMessageComponent`, it might be better to keep this type as an unexported internal implementation detail, then use an `UnmarshalJSON` function on `Message`; this doesn't really seem like something that we'd want to expose to the library's interface.

Reply:
hmm, thought about that, but yeah, thanks